pandas/tests/frame/test_nonunique_indexes.py

```diff
@@ -513,3 +513,10 @@ def test_set_value_by_index(self):
 
         df.iloc[:, 0] = 3
         tm.assert_series_equal(df.iloc[:, 1], expected)
+
```

Comment:
can you construct this exactly as in the OP; e.g. we want these converted to numpy *before* the constructor. also pls parameterize as there are multiple tests indicated

Reply:
Hmm, I am not sure I follow, what would be a good way to parameterize in this scope?

I saw that in order functions we have access to a `check` function, but if I were to put my test inside another function, it would probably be the `test_columns_with_dups` function, which has no access to this particular mechanism.

As for the numpy parts, I just added a commit to address that, maybe it is as it should be, thanks for the review!

Reply:
@jreback I think I misread your request the first time, you are saying there should be other cases for this test? Looking at the issue again it seems to me that this single test should be enough